main.cpp

```diff
@@ -305,13 +247,36 @@ struct DoubleType
     DoubleType& multiply(double rhs );
     DoubleType& divide(double rhs );
     
+    double pow(const IntType&);
+    double pow(const FloatType&);
+    double pow(const DoubleType&);
+    double pow(double);
+    
+    double powInternal(double* val, double arg);
+    
+    
 private:
     double* value;
+    
 };
+
 struct IntType
 {
     IntType(int intIn_) : value(new int(intIn_)){}
     
+    IntType(const IntType& other)
+    {
+        value = new int();
+        *value = *other.value;
+    }
+    
+    IntType& operator= (const IntType& other)
+    {
+        value = new int();
+        *value = *other.value;
+        return *this;
+    }
```

Comment:
You do not need these Rule of 3 functions, and they are implemented incorrectly.  You have a leak!
- [ ] remove these